test/algorithms/simulation/sidb/operational_domain.cpp

```diff
@@ -0,0 +1,438 @@
+
+            // for the selected range, all samples should be within the parameters and operational
+            check_op_domain_params_and_operational_status(op_domain, op_domain_params,
+                                                          operational_domain::operational_status::OPERATIONAL);
+
+            CHECK(mockturtle::to_seconds(op_domain_stats.time_total) > 0.0);
+            CHECK(op_domain_stats.num_simulator_invocations == 200);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations == 100);
+            CHECK(op_domain_stats.num_operational_parameter_combinations == 100);
+            CHECK(op_domain_stats.num_non_operational_parameter_combinations == 0);
+        }
+        SECTION("random_sampling")
+        {
+            const auto op_domain =
+                operational_domain_random_sampling(lyt, create_id_tt(), 100, op_domain_params, &op_domain_stats);
+
+            // check if the operational domain has the correct size (max 10 steps in each dimension)
+            CHECK(op_domain.operational_values.size() <= 100);
+
+            // for the selected range, all samples should be within the parameters and operational
+            check_op_domain_params_and_operational_status(op_domain, op_domain_params,
+                                                          operational_domain::operational_status::OPERATIONAL);
+
+            CHECK(mockturtle::to_seconds(op_domain_stats.time_total) > 0.0);
+            CHECK(op_domain_stats.num_simulator_invocations <= 200);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations <= 100);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations > 0);
+            CHECK(op_domain_stats.num_operational_parameter_combinations <= 100);
+            CHECK(op_domain_stats.num_non_operational_parameter_combinations == 0);
+        }
+        SECTION("flood_fill")
+        {
+            const auto op_domain =
+                operational_domain_flood_fill(lyt, create_id_tt(), 1, op_domain_params, &op_domain_stats);
+
+            // check if the operational domain has the correct size (max 10 steps in each dimension)
+            CHECK(op_domain.operational_values.size() == 100);
+
+            // for the selected range, all samples should be within the parameters and operational
+            check_op_domain_params_and_operational_status(op_domain, op_domain_params,
+                                                          operational_domain::operational_status::OPERATIONAL);
+
+            CHECK(mockturtle::to_seconds(op_domain_stats.time_total) > 0.0);
+            CHECK(op_domain_stats.num_simulator_invocations == 200);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations == 100);
+            CHECK(op_domain_stats.num_operational_parameter_combinations == 100);
+            CHECK(op_domain_stats.num_non_operational_parameter_combinations == 0);
+        }
+        SECTION("contour_tracing")
```

Comment:
In rare cases (~10 %), I get ``Assertion failed: (std::find(neighborhood.cbegin(), neighborhood.cend(), backtrack) != neighborhood.cend() && "The backtrack point must be part of the neighborhood"), function operator(), file operational_domain.hpp, line 448.``

Reply:
Huh curious :thinking: Will investigate. Thank you!

Reply:
I've found the source of the problem and fixed it. Many thanks for reporting! :pray: